// async-validator/es/validator/index.js
import string from 'async-validator/lib/validator/string';
import method from 'async-validator/lib/validator/method';
import number from 'async-validator/lib/validator/number';
import boolean from 'async-validator/lib/validator/boolean';
import regexp from 'async-validator/lib/validator/regexp';
import integer from 'async-validator/lib/validator/integer';
import float from 'async-validator/lib/validator/float';
import array from 'async-validator/lib/validator/array';
import object from 'async-validator/lib/validator/object';
import enumValidator from 'async-validator/lib/validator/enum';
import pattern from 'async-validator/lib/validator/pattern';
import date from 'async-validator/lib/validator/date';
import required from 'async-validator/lib/validator/required';
import type from 'async-validator/lib/validator/type';

export default {
  string: string,
  method: method,
  number: number,
  boolean: boolean,
  regexp: regexp,
  integer: integer,
  float: float,
  array: array,
  object: object,
  'enum': enumValidator,
  pattern: pattern,
  date: date,
  url: type,
  hex: type,
  email: type,
  required: required
};
